src/utils/url.ts

```diff
@@ -155,3 +156,59 @@ export function isURLEnabled(url: string, userSettings: UserSettings, {isProtect
     }
     return (!isInDarkList && !isURLInUserList);
 }
+
+function getExtensionPageObject(path: string): Promise<chrome.windows.Window> | Promise<chrome.tabs.Tab> {
```

Comment:
Let's put this function into `src/ui/utils.ts`, it is more related to UI than to URL, utilities in root folder should be related to both UI and background.